pallets/gas/src/property_tests.rs

```diff
@@ -0,0 +1,212 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Properties and invariants that are checked:
+//! 1. all non-external nodes have a parent in GasTree storage.
+//! 2. all non-external nodes have at least one ancestor with value (i.e., ValueNode::node_with_value procedure always return Ok),
+//! however this value can be equal to 0. Also there are no guarantees that this ancestor is a parent.
+//! 3. all nodes can't have consumed parent with zero refs (there can't be any nodes like that in storage) between calls to `ValueTree::consume`.
+//! Therefore, if node is deleted, it is consumed and has zero refs (and zero value, if it is able to hold value and is of `ValueType::SpecifiedLocal` type).
+//! So if there is an existing consumed node, it has non-zero refs counter and a value >= 0 (between calls to `ValueTree::consume`)
```

Comment:
I suggest leaving it as it is and renaming everything in #1072, because such refactoring will lead to renaming entities.
@gshep can I ask you please to pay attention to that?

Reply:
> I suggest leaving it as it is and renaming everything in #1072, because such refactoring will lead to renaming entities.
> @gshep can I ask you please to pay attention to that?

Yeah, sure, any our conversations about renaming should be done in separate PR with huge refactoring
But at this comment, I mean, that we have `trait ValueTree`, but you name it differently in docs: `ValueTree` and `GasTree` - that's bad

Btw, I vote for renaming to `GasTree` and `GasNode` for future

Reply:
done